### ISSUE TITLE
[BISECTED] man page group header is overwritten by detailed description

### ISSUE THREAD

**Issue by duncan-roe** [doxygen/doxygen#7901]
IN BRIEF
===
When generating a man page for a module, doxygen dumps the Detailed Description
where the Brief Description (title line) should be.

IN DETAIL
===
Samples
---
This code

    /**
     * \defgroup baz BAZ TITLE
     *
     * This is \b not the title line.
     * \n
     * Generated by doxygen 1.8.13 (d4beb392b452d14d375e1406ec4fb0f6436081ee)
     *
     * @{
     */
         
    /**
     * foo - a sample function
     * \param bar First parameter of foo
     */
    void foo(int bar)
    {
      return;
    }
        
    /**
     * @}
     */
should produce this result

    NAME
           baz - BAZ TITLE
    
       Functions
           void foo (int bar)
    
    Detailed Description
           This is not the title line.
           Generated by doxygen 1.8.13 (d4beb392b452d14d375e1406ec4fb0f6436081ee)
(and so on)  
but produces 

    NAME
           baz - This is not the title line.
           Generated by doxygen 1.8.19 (cbf83033232f8c76836bd8f086bfad146ed74e00)
    
    
    SYNOPSIS
       Functions
           void foo (int bar)
    
    Detailed Description
           This is not the title line.
           Generated by doxygen 1.8.19 (cbf83033232f8c76836bd8f086bfad146ed74e00)
(and so on)

Bisection History
---
The last commit that produced the wanted result was d4beb392 as above.
At the next commit, the title line was simply not there:

    NAME
           baz
    
    SYNOPSIS
       Functions
           void foo (int bar)
    
    Detailed Description
           This is not the title line.
           Generated by doxygen 1.8.13 (98d3f8e7d581c589e7bd1a7faf98fc6736847cf1)
(and so on)  
This continued until commit 0ef84d5a which behaves the same as the most
recent commit (above).

To Reproduce
---
Put
[try\_commit.txt](https://github.com/doxygen/doxygen/files/4901839/try_commit.txt)
somewhere in your PATH, remove the .txt suffix and make it executable.  
Now if you enter `try_commit -h` you should see:

    Check out, build & check operation of 'man' for the given doxygen git revision
    Usage:
    try_commit [options...] [<tree-ish>]
        Checkout and build the specified revision.
        Run the built doxygen to make man pages.
        Run man on the made man pages.
    
    Options:
        -o name: Omit 'man ../.../name.3'. May be given more than once.
                 meta-name "all" omits all man pages.
        -m     : Skip "Checkout and build ..." step above.
        -i name: Include 'man ../.../name.3'. May be given more than once.
                 meta-name "all" includes all man pages (the default behaviour).
                 (e.g. "try_commit -m -o all -i baz" shows only the baz page).
        -h     : Show help.
        -c     : Skip checkout but do build (e.g. during git bisect).
    
    Environment Variables:
        MAN_ONLY     - set "true" for same effect as -m
        GIT_CHECKOUT - set "false" for same effect as -c
        BAZ          - set "true" for same effect as -o baz
        QUX          - set "true" for same effect as -o qux
        WALDO        - set "true" for same effect as -o waldo
        NUMJOBS      - override the default number of 'make' jobs e.g. "-j 1"

For _this_ bug report, only baz.3 is of interest.
So to build 0ef84d5a, use  
`try_commit -o all -i baz 0ef84d5a 2>/dev/null | tail -n28`  
**WARNING:** try_commit can check out arbitrary git revisions so you may prefer
to run it in a git clone (which nobody else is trying to use).

**Comment by albert-github** [comment]
As `try_commit` can check out arbitrary git revision it is better to to split the example showing the problem and the git code part.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/4903219/example.tar.gz)

**Comment by albert-github** [comment]
Looking at the given commits (thanks):
- 0ef84d5a
  This is the commit: 
  > Date:   Thu Jun 13 13:13:04 2019 +0200
  >
  >  Missing brief descriptions with `\defgroup`
  >
  >  Due to the fact that issue #6945 ended a brief description, all following was put into the detailed description (unless explicit \brief was used). The `\defgroup` should end the current brief description, but first line of the `\defgroup` should be seen as a new brief description.
  >  Problem can be seen with in the doxygen documentation in the example in the Grouping part and also with #7053 a small example is given in the comment.

  So this is introduce for issue #6945 and its pull request #6971, also related to #7054, #7852 and #7867.
- d4beb392
   the next commit I see is 02fb542:
   > Date: Monday, September 19, 2016 8:43:58 PM
   >
   > Bug 771344 - Class name 'internal' breaks class hierarchy in C++

   where the change is:
   ```
   <BasesProt>"internal"{BN}+              { lineCount(); baseProt = Package; }
  <BasesProt>"internal"{BN}+              { if (!insideCli) REJECT ; lineCount(); baseProt = Package; }
   ```
   but I cannot relate this change to the mentioned changes.
   - Is the mentioned hash (d4beb392) correct?

**Comment by duncan-roe** [comment]
@arnld-github Sorry `d4beb39` is a merge commit and I should not have quoted it. The failing commit is `98d3f8e7`.  Observe this revision in the posted failing man page: `try_commit` arranges generated C source to display numeric `doxygen --revision` and commit hash even for old revs where `doxygen --revision` didn't do this for you.

Using `git log --no-merges` I see the previous commit is `84017cac` which doesn't show the bug. Tested by:

    git checkout 84017cac3bc9d08aa95a90cf0e44b913c492c939
    try_commit -c -o all -i baz 2>/dev/null | tail -n28
    finished...
    baz(3)                            My Project                            baz(3)
    
    
    
    NAME
           baz - BAZ TITLE
    
       Functions
           void foo (int bar)
    
    Detailed Description
           This is not the title line.
           Generated by doxygen 1.8.13 (84017cac3bc9d08aa95a90cf0e44b913c492c939)
    
    Function Documentation
       void foo (int bar)
           foo - a sample function
    
           Parameters:
               bar First parameter of foo
    
    Author
           Generated automatically by Doxygen for My Project from the source code.
    
    
    
    Version 1.8.13                  Sat Jul 11 2020                         baz(3)

With regard to your suggested `example.tar.gz`: I hope you can now see more of why `try_commit`  is needed as_is (the ability to check out commits is just a convenience: I could remove that and make `-c` the default); or I can elaborate further if you prefer.

**Comment by albert-github** [comment]
- what is `doxygen --revision`
- how doxygen is build is depending on the development environment one has uses. For me it won't directly work (Windows) on Cygwin it would. Also mixing a model / example with build code is not always a good idea (and not directly reusable), but this is not the problem / issue here. The "example.tar.gz" should be enough to show the problem (with of course the mentioned bisected commit IDs).

**Comment by duncan-roe** [comment]
*what is `doxygen --revision`*
It's documented in https://www.doxygen.nl/manual/trouble.html#bug_reports to output the version of doxygen you are using. But it only seems to work in recent versions, while **`doxygen -v`** always works, so I'll use that in future

**Comment by albert-github** [comment]
Unfortunately you misread it is nor `--revesion` but `--version`.

**Comment by duncan-roe** [comment]
My bad - I actually have `--version` in `try_commit` :-/

**Comment by duncan-roe** [comment]
*how doxygen is build is depending on the development environment*
Ok with that - will send tar.gz in future

**Comment by duncan-roe** [comment]
Just to be clear, there are 2 phases to this bug. At commit 98d3f8e7, the brief description disappeared. It was still there at commit 84017cac. That was phase 1.
At commit 0ef84d5, the detailed description was inserted where the brief description should be (the preceding commit was 2934e3cb). This is phase 2.
Perhaps phase 2 is not a bug at all, but just a consequence of doxygen's not realising there is a Brief Description owing to the phase 1 bug.

**Comment by duncan-roe** [comment]
If you checkout the failing 98d3f8e, the previous non-merge commit shows as 78b43ec. 78b43ec displays the title line correctly and has a much smaller diff with 98d3f8e than 84017ca has.

**Comment by duncan-roe** [comment]
@albert-github : I developed a patch which solves this to my satisfaction and posted it to doxygen-develop so you could use `git am` to apply it.
If you prefer, I can post it here. The commit message is

src: fix bug #7901

Modified:

 src/commentscan.l: 0ef84d5 duplicated Detailed Description

 src/groupdef.cpp: 98d3f8e stopped outputting Brief Description

 src/mangen.cpp: 98d3f8e stopped outputting minus sign before B.D.

**Comment by albert-github** [comment]
It is better to make a pull request for it, though there are indeed 2 phases as you mentioned and also the switch regarding the brief / detailed description (as also indicated in https://github.com/doxygen/doxygen/issues/7901#issuecomment-656673444 and mentioned in other issues).

**Comment by duncan-roe** [comment]
Sure, I can fork this repo and do it that way. Tomorrow - it's late now.

**Comment by doxygen** [comment]
This is a specific occurrence of the more general issue also mentioned in #7852, and #7867, dealing with how `JAVADOC_AUTOBRIEF` should be handled for comment blocks which include 'structural commands' like `\defgroup`. The current implementation will end the brief description, but this is not always desired or intuitive. 

We should fix it structurally instead of making small fixes for individual situations which will break other things and make things inconsistent. 

I'll look into it.

**Comment by albert-github** [comment]
A patch (commit: 327423e217337bf4e64515aba6cf78b9877bc165) has been integrated into the master version.

**Comment by duncan-roe** [comment]
I am seeing two minus signs in title line now

    NAME
           baz -  - BAZ TITLE
    SYNOPSIS

**Comment by duncan-roe** [comment]
New example archive with and without title lines
[example.tar.gz](https://github.com/doxygen/doxygen/files/4940923/example.tar.gz)

**Comment by albert-github** [comment]
@duncan-roe 
Is this correct or what are the problems?

I do get 2 warnings:
- `.../bar.c:1: warning: missing title after \defgroup qux`
- `.../bar.c:2: warning: found \endinternal without matching \internal`

Note: the doxygen versions in the example are, in my opinion, a bit misleding as they are from an older version.

**Comment by duncan-roe** [comment]
The warnings in bar.c are expected (except the line numbers are wrong, that is to be the subject of a later bug report, not as important as this one). The point of bar.c is to show 2 minus signs in  the title line you get when there is no brief description; foo.c shows 2 minus signs as well.
Sorry about the doxygen versions in the attachment: I will edit them [out] in future.

**Comment by doxygen** [comment]
@duncan-roe Looks like you see the problem only after applying your patch on top of the current master, but that should no longer be needed. You added the '-' in `mangen.cpp` while I added it in `groupdef.cpp` so if you combine boths you get the double minus which you fixed by removing the one from `groupdef.cpp` again, but by doing so you will get cases where there is a minor without any content.

**Comment by duncan-roe** [comment]
@doxygen Indeed. I should have kept my clone of this repo rather than thinking github could patch my fork back to being in sync (first time doing this).

**Comment by duncan-roe** [comment]
@doxygen Now this is fixed, please can you make a release? [libnetfilter_queue-1.0.5](https://netfilter.org/projects/libnetfilter_queue/downloads.html) can be configured to use doxygen to build a full set of man pages but I have been holding back on [recommending this for the next Slackware release](https://www.linuxquestions.org/questions/slackware-14/requests-for-current-14-2-15-0-a-4175620463/) because of this bug.
Happy to close this issue otherwise - do I need to do that?

**Comment by albert-github** [comment]
There is no need to close the issue as this will be done at the moment of an official release (as it signals a problem that has been fixed for the upcoming release).
The proposed pull request #7910 and #7909 have already been closed.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
